test/Test/Time/Units.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE DataKinds          #-}
+{-# LANGUAGE ExplicitNamespaces #-}
+{-# LANGUAGE TypeApplications   #-}
+{-# LANGUAGE TypeOperators      #-}
+
+module Test.Time.Units
+       ( runTests
+       ) where
+
+import Data.Ratio ((%))
+import Test.Tasty (TestTree, defaultMain)
+import Test.Tasty.Hspec (Spec, describe, it, shouldBe, testSpec)
+
+import Time (Day, DayUnit, Fortnight, MicroSecond, MicroSecondUnit, MilliSecond, MilliSecondUnit,
+             Second, SecondUnit, Time (..), WeekUnit, convertUnit)
+
+runTests :: IO ()
+runTests = do
+    tests <- specTests
+    defaultMain tests
+
+specTests :: IO TestTree
+specTests = testSpec "Units" spec_convertUnit
+
+spec_convertUnit :: Spec
+spec_convertUnit =
+    describe "Unit Conversion Test" $ do
+        it "11 seconds is 11000 milliseconds" $
+            convertUnit @MilliSecondUnit (Time 11 :: Second) `shouldBe` Time 11000
```

Comment:
You can write just `11000` instead of `Time 11000` and `11 :: Second` instead of `Time 11 :: Second`.
Another possible approach is to write `convertUnit @MilliSecondUnit @SecondUnit 11`. Chose which one you like more :)